base/src/test/java/io/spine/base/IdentifierTest.java

```diff
@@ -0,0 +1,386 @@
+        void ofWrappedString() {
+
+            StringValue id = toMessage(TEST_ID);
+
+            String result = Identifier.toString(id);
+
+            assertEquals(TEST_ID, result);
+        }
+
+        @Test
+        @DisplayName("Message with string field")
+        void ofMessage() {
+            StringValue id = toMessage(TEST_ID);
+
+            String result = Identifier.toString(id);
+
+            assertEquals(TEST_ID, result);
+        }
+
+        @Test
+        @DisplayName("Message with nested Message")
+        void ofNestedMessage() {
+            StringValue value = toMessage(TEST_ID);
+            NestedMessageId idToConvert = NestedMessageId
+                    .newBuilder()
+                    .setId(value)
+                    .build();
+
+            String result = Identifier.toString(idToConvert);
+
+            assertEquals(TEST_ID, result);
+        }
+
+        @Test
+        @DisplayName("Any")
+        void ofAny() {
+            StringValue messageToWrap = toMessage(TEST_ID);
+            Any any = AnyPacker.pack(messageToWrap);
+
+            String result = Identifier.toString(any);
+
+            assertEquals(TEST_ID, result);
+        }
+    }
+
+    @Test
+    @DisplayName("provide string conversion of messages with several fields")
+    void toStringSeveralFields() {
+        String nestedString = "nested_string";
+        String outerString = "outer_string";
+        Integer number = 256;
+
+        StringValue nestedMessageString = toMessage(nestedString);
+        SeveralFieldsId idToConvert = SeveralFieldsId.newBuilder()
+                                                     .setString(outerString)
+                                                     .setNumber(number)
+                                                     .setMessage(nestedMessageString)
+                                                     .build();
+
+        String expected =
+                "string=\"" + outerString + '\"' +
+                        " number=" + number +
+                        " message { value=\"" + nestedString + "\" }";
+
+        String actual = Identifier.toString(idToConvert);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName(NOT_ACCEPT_NULLS)
+    void nullCheck() {
+        new NullPointerTester()
+                .testAllPublicStaticMethods(Identifier.class);
+    }
+
+    @Test
+    void create_values_depending_on_wrapper_message_type() {
+        assertEquals(10, Type.INTEGER.fromMessage(toMessage(10)));
+        assertEquals(1024L, Type.LONG.fromMessage(toMessage(1024L)));
+
+        String value = getClass().getSimpleName();
+        assertEquals(value, Type.STRING.fromMessage(toMessage(value)));
+    }
+
+    @Nested
+    @DisplayName("reject unsupported")
+    class RejectUnsupported {
+
+        @Test
+        @DisplayName("value")
+        @SuppressWarnings("UnnecessaryBoxing") // We want to make the unsupported type obvious.
```

Comment:
It's interesting that the suppression is the same, but the comment is different from the comments above.